core/server/worker/src/main/java/alluxio/worker/netty/BlockReadHandler.java

```diff
@@ -101,21 +103,38 @@ protected DataBuffer getDataBuffer(BlockReadRequestContext context, Channel chan
       openBlock(context, channel);
       BlockReader blockReader = context.getBlockReader();
       Preconditions.checkState(blockReader != null);
-      if (mTransferType == FileTransferType.TRANSFER
-          && (blockReader instanceof LocalFileBlockReader)) {
+      DataBuffer dataBuffer = null;
+      if (mTransferType == FileTransferType.TRANSFER) {
```

Comment:
Is this check still relevant? Because even if not using zero-copy, we are still using `transferTo`instead of `read`.